### PR TITLE
[chore] add text about unmaintained vendor components

### PR DESCRIPTION
**Pull request by TylerHelmuth**
#### Description

This PR adds language to our definition of `unmaintained` that allows vendor-specific components to become unmaintained if they lose an active code owner from the contributing vendor. This is necessary to prevent a vendor-specific component form being maintained only by the community instead of by the vendor.  Since vendors had privileged access to getting components accepted in contrib they must be held to a higher standard for maintaining those components.

### CODE REVIEW

**Comment by atoulme** [comment]
How do we know if a component is vendor-specific? Or maybe over time, do we want to no longer have any vendor-specific components?

**Review by atoulme — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
> How do we know if a component is vendor-specific? Or maybe over time, do we want to no longer have any vendor-specific components?

Good question. The git history/issues would have it, but right now it is probably relying more on the approvers/maintainers that were assigned as part of the rotation.

**Review by songy23 — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Just to make is more interesting, is "datadogreceiver" vendor specific?

**Comment by TylerHelmuth** [comment]
@bogdandrutu I'd consider a component in contrib `vendor-specific` if during the component proposal process the `vendor-specific` checkbox was checked and the `vendor-specific component` label was used ((implying the vendor submitted the proposal) and it was accepted under the grounds of "yes we have to accept this so that we are playing fair with all vendors".

The datadogreceiver was not contributed under the above measures (and there are a few other components in contrib that touch vendor things but were not contributed by that vendor) so ! would say it, and the others, are not vendor specific.

I think to merge this change it would be appropriate to classify the components in contrib that are subject to the additional guidelines this PR proposes. I could do that by adding something to the metadata.yaml for the components or by building some list somewhere that indicates the vendor-specific components contrib houses.

File: docs/component-stability.md [hunk 1]
```diff
@@ -73,3 +73,8 @@ The component is planned to be removed in a future version and no further suppor
 ### Unmaintained
 
 A component identified as unmaintained does not have an active code owner. Such component may have never been assigned a code owner or a previously active code owner has not responded to requests for feedback within 6 weeks of being contacted. Issues and pull requests for unmaintained components will be labelled as such. After 6 months of being unmaintained, these components will be removed from official distribution. Components that are unmaintained are actively seeking contributors to become code owners.
+
+Components that were accepted based on being vendor-specific components will be marked as unmaintained if
+it has no active code owners from the vendor even if there are other code owners listed. Other active code
```
Comment (mx-psi) [inline docs/component-stability.md:78]:
```suggestion
they have no active code owners from the vendor even if there are other code owners listed. Other active code
```

File: docs/component-stability.md [hunk 2]
```diff
@@ -73,3 +73,8 @@ The component is planned to be removed in a future version and no further suppor
 ### Unmaintained
 
 A component identified as unmaintained does not have an active code owner. Such component may have never been assigned a code owner or a previously active code owner has not responded to requests for feedback within 6 weeks of being contacted. Issues and pull requests for unmaintained components will be labelled as such. After 6 months of being unmaintained, these components will be removed from official distribution. Components that are unmaintained are actively seeking contributors to become code owners.
+
+Components that were accepted based on being vendor-specific components will be marked as unmaintained if
```
Comment (mx-psi) [inline docs/component-stability.md:77]: Should we say that we will attempt to reach out to the vendor to notify them of the component being marked as unmaintained?

**Review by bogdandrutu — APPROVED** [review]

**Comment by atoulme** [comment]
@TylerHelmuth please resolve conflicts

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
This looks ok to me, conflicts need resolving though

**Review by codeboten — APPROVED** [review]

File: docs/component-stability.md (hunk 2 above)
Comment (TylerHelmuth) [reply docs/component-stability.md:77]: Added

**Review by mx-psi — APPROVED** [review]

**Pull request merged**